Eee status display

logicog opened the pull request:
Adds an EEE status command and EEE status and configuration web-pages

The following is the eee status command:
```
> eee status
Port: 5: SFP
Port: 1: Advertising:        1G  100M   Link Partner:  2.5G  1G  100M INACTIVE 
Port: 2: Advertising:        1G  100M   Link Partner:  2.5G  1G  100M INACTIVE 
Port: 3: Advertising:        1G  100M   Link Partner:        1G  100M ACTIVE   
Port: 4: Advertising:        1G  100M   Link Partner:                 INACTIVE 
Port: 6: SFP

> eee on

> eee status
Port: 5: SFP
Port: 1: Advertising:  2.5G  1G  100M   Link Partner:  2.5G  1G  100M ACTIVE   
Port: 2: Advertising:  2.5G  1G  100M   Link Partner:  2.5G  1G  100M ACTIVE   
Port: 3: Advertising:  2.5G  1G  100M   Link Partner:        1G  100M ACTIVE   
Port: 4: Advertising:  2.5G  1G  100M   Link Partner:                 INACTIVE 
Port: 6: SFP

> eee off

> eee status
Port: 5: SFP
Port: 1: Advertising:                   Link Partner:  2.5G  1G  100M INACTIVE 
Port: 2: Advertising:                   Link Partner:  2.5G  1G  100M INACTIVE 
Port: 3: Advertising:                   Link Partner:        1G  100M INACTIVE 
Port: 4: Advertising:                   Link Partner:                 INACTIVE 
Port: 6: SFP
```

This is the EEE html interface, admittedly this lacks a good design and taste concerning colors:
<img width="1883" height="761" alt="Screenshot_20251018_095212" src="https://github.com/user-attachments/assets/49d5d7c1-17d8-4b49-8c87-e5c843647b4f" />

vDorst commented:
On my pc with firefox and brave it looks like this.

<img width="1273" height="422" alt="Screenshot_20251019_223159" src="https://github.com/user-attachments/assets/25dd0d48-35c3-4b19-8085-437ccffc6ac4" />


I wonder maybe just use binary values in the low layer/json.
So we need one byte per port.

vDorst commented on httpd/page_impl.c:
```diff
@@ -208,6 +210,76 @@ void send_counters(char port)
+			else
+				char_to_html('0');
+			slen += strtox(outbuf + slen, "\", \"active\": ");
+			reg_read_m(RTL8373_PHY_EEE_ABLTY);
```
Maybe just read this value outside the loop, higher up in the code.
It is just one byte to cache.

vDorst commented on html/eee_sub.js:
```diff
@@ -0,0 +1,18 @@
+    cmd = cmd + "off";
+  console.log("eeeSub port " + port, ", value " + enable);
+  try {
+    const response = await fetch('/cmd', {
```
Javascript keeps sending request, but doesn't wait for an answer, before starting a new one. Every request seems to stackup and ddossing the device.

logicog commented:
Added timeouts to the XhttpGet requests. They were piling up because they were still issued without a timeout while the device was reconfiguring the interfaces and going nowhere.
Also, I moved the execution of the command into a deferred part of the idle() loop. Also for the console.
I no longer see the Javascript DDOSing the switch. Also, all the issues identified above have been addressed.

vDorst commented:
Looks good but.... now the `Port Statistics` site is broken because of the high response-time and the 1500mS timeout.

<img width="1121" height="211" alt="port" src="https://github.com/user-attachments/assets/fd9a17f3-4a15-4d42-88a6-7818c3ca2481" />
